main.cpp

```diff
@@ -64,151 +63,592 @@ New/This/Pointers/References conclusion
 
 #include <iostream>
 
+struct DoubleType;
+struct IntType;
+
 struct FloatType
 {
-    float add(float lhs, float rhs);
-    float subtract(float lhs, float rhs);
-    float multiply(float lhs, float rhs);
-    float divide(float lhs, float rhs);
+    FloatType() {}
```

Comment:
You should not need this default constructor

- [ ] remove default constructor